types/detect/detect.go

```diff
@@ -94,3 +94,18 @@ type DataSource interface {
 
 var ErrDataNotFound = errors.New("requested data was not found")
 var ErrKeyNotSupported = errors.New("queried key is not supported")
+var ErrFailedToUnmarshal = errors.New("given value could not be unmarshaled")
+
+type WriteableDataSource interface {
```

Comment:
perhaps rename to `DataSourceWriter`? I think `Writer` instead of `Writeable` matches golang naming better.

Reply:
I think if we do that then `DataSource` should be changed to `DataSourceReader` too no? But I don't think so anyway because a `WritableDataSource` necessarily has to implement the full `DataSource` interface, and not just be able to write into one. That's why the name conveys additional behavior and not a functionality (this data source is _also_ writable as opposed to this is the _method_ to write to a particular data source).